### DISCUSSION TITLE
Interest in a Moderator Role UI component?

### DISCUSSION THREAD

**Discussion by byjrack** [drodil/backstage-plugin-qeta#71] [Ideas]
We were trying to think about how to scale Tag governance a bit better. It's either a bit of anarchy or we use the app-config which works for a handful, but may get a bit crazy if you have a reasonable about of Tags.  Not to mention restarting the instance to get a new Tag is not ideal.

Figured a middle path is to have trusted moderators able to create Tags.  This creates some friction in the system, but a loose taxonomy can fall apart quickly.  Need to figure out the feedback loop for the moderators though.  Could be a UI component with RBAC to do some of these basics.

Also had thought about an external git process that would directly insert into the database as well, but trying not to go too out of band.

Does anyone else have creative approaches to Tag governance to avoid all the permutations and regular cleanup/consolidation needed to keep things discoverable?

**Comment by drodil** [comment]
I have been thinking about moderator role that could edit and delete questions/answers that are not their own. Managing tags could also be part of that role. And it should use the Backstage permission framework to distinguish the moderators from normal users. This requires some changes to the current permission checks that are done all the way in the database level + UI changes for moderating the tags etc. If there is someone interested to contribute to this, it would be much appreciated.

**Comment by drodil** [comment]
@byjrack hi again! There's now a moderator panel in the plugin and implementing the possibility for moderators to manage tags would be possible now. I've added it to backlog #241 and maybe have time to work on it sometime soon!

**Reply by drodil** [reply]
This is now available in 3.15.0. Basically, this means that you can set the config `qeta.tags.allowCreation` to false and moderators can create + delete tags via UI.

**Reply by byjrack** [reply]
Awesome.  With your notifications work in Backstage I am trying to nudge folks to look at the Q&A system we landed on to see if they want to move folks again.  You can guess there is plenty of resistance there, but trying to make the business case for consolidation to Backstage as one of those simplify and standardize plays.
